python/ray/train/v2/_internal/execution/context.py

```diff
@@ -133,6 +137,13 @@ def get_synchronization_actor(self):
     def get_checkpoint(self):
         return self.checkpoint
 
+    def get_all_reported_checkpoints(self) -> List[ReportedCheckpoint]:
+        return ray.get(
+            self.controller_actor.get_all_reported_checkpoints.remote(
+                self.num_reported_checkpoints
+            )
```

Comment:
this is accessing an optional controller actor

Reply:
problem:
* `ray.tune.report` needs to populate a local queue on the Tune Trainable actor
* Right now, the controller runs on that Trainable actor, so it will call `ray.tune.report` from the `after_report` callback.
* If the controller is a separate actor, then `ray.tune.report` doesn't work.

So we might need a better way to bridge TrainController and the Tune Trainable actor, so that we can always have a `TrainController` actor when running Ray Train.